### PR TITLE
Benchmarks: Build Pipeline - fix nccl and nccl test version to 2.18.3 to resolve hang issue in cuda12.2 docker

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
Upgrade nccl tests to resolve all gather hang issue in cuda12.2 docker.

### CODE REVIEW

**Comment by yzygitzh** [comment]
Is this hanging issue with nccl or msccl? For msccl maybe it's better to build and use msccl/tests/msccl-tests-nccl

**Review by cp5555 — APPROVED** [review]

**Comment by cp5555** [comment]
What's the version for NCCL test?

**Comment by yukirora** [comment]
> What's the version for NCCL test?

[v2.13.8](https://github.com/NVIDIA/nccl-tests/releases/tag/v2.13.8)

**Review by yzygitzh — APPROVED** [review]

File: dockerfile/cuda12.2.dockerfile [hunk 1]
```diff
@@ -113,6 +113,13 @@ RUN cd /tmp && \
     mv amd-blis /opt/AMD && \
     rm -rf aocl-blis-linux-aocc-4.0.tar.gz
 
+# Install NCCL 2.18.3
+RUN cd /tmp && \
+    git clone -b 2.18.3-1 https://github.com/NVIDIA/nccl.git && \
```
Comment (abuccts) [inline dockerfile/cuda12.2.dockerfile:118]: can you update the versions at the beginning of this file as well?

**Review by abuccts — APPROVED** [review]

**Pull request merged**